lib/Doctrine/ORM/Internal/Hydration/ArrayHydrator.php

```diff
@@ -124,9 +124,7 @@ protected function hydrateRowData(array $row, array &$result): void
                                 $baseElement[$relationAlias][] = $element;
                             }
 
-                            end($baseElement[$relationAlias]);
-
-                            $this->_identifierMap[$path][$id[$parent]][$id[$dqlAlias]] = key($baseElement[$relationAlias]);
+                            $this->_identifierMap[$path][$id[$parent]][$id[$dqlAlias]] = array_key_last($baseElement[$relationAlias]);
```

Comment:
This causes a new psalm error, that we baselined, because we don't think it's legitimate. But please triple check.